pkg/karmadactl/cmdinit/kubernetes/deploy_test.go

```diff
@@ -172,8 +172,9 @@ func TestInitKarmadaAPIServer(t *testing.T) {
 
 	// Create a new CommandInitOption
 	initOption := &CommandInitOption{
-		KubeClientSet: clientset,
-		Namespace:     "test-namespace",
+		KubeClientSet:             clientset,
+		Namespace:                 "test-namespace",
+		WaitComponentReadyTimeout: 200,
```

Comment:
Seems like it is the reason of failed.  0 by default.

Reply:
need to discussion: change the default value of WaitComponentReadyTimeout and some usage of PollUntilContextTimeout.

Reply:
I check the code, PollUntilContextTimeout should not change the default value of `WaitComponentReadyTimeout`

Reply:
@jwcesign 

I think you are right after i do the double check.

I made more test and seems like it's the bug of util.wait, it's can fix with bump k8sio dependenic to v0.27.4.

the only new commit for util.wait is https://github.com/kubernetes/kubernetes/commit/59cd1d0b3bb378f40a639e21b615f4df1d4a5a14

I might suggest that we first use util.wait in our tests and see how it goes after a while.

@RainbowMango   Do you have any concerns about this if i submit a PR to update the dependency to 27.4?

Note: this PR update dependencies is working for test.

Reply:
We still need to wait for https://github.com/kubernetes/kubernetes/pull/119762, right? It isn't included in v0.27.4.

Reply:
Yes, i missed it. And other thing is Kind have not `v0.27.4` yet

Reply:
Why do we rely on kind@v0.27.4? We are relying the dependencies, but we are testing against [Kubernetes v1.27.4](https://github.com/karmada-io/karmada/blob/25db1e06d033e011bbb2e81bcdeb24b477277671/hack/local-up-karmada.sh#L51).